src/blob/index/header.rs

```diff
@@ -39,13 +42,15 @@ impl IndexHeader {
         meta_size: usize,
         hash: Vec<u8>,
         blob_size: u64,
+        key_size: u16,
```

Comment:
Different order of parameters in `with_hash` and in `new`. This increases the chance of error. Place `key_size` before `blob_size` as it is in `new` function